### PR TITLE
[playgrounds]: implement "Add library" command

### PR DESCRIPTION
**Pull request by legomushroom**
The PR implements "Add library" command:

playgrounds-add-library

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -719,7 +737,13 @@
           "icon": "images/activity-bar-icon.svg"
         }
       ]
-    }
+    },
+    "jsonValidation": [
+      {
+        "fileMatch": "playground.json",
+        "url": "https://gist.githubusercontent.com/legomushroom/e6ef081dc3f466b6228acac5ee9b620d/raw/gistpad-schema.json"
```
Comment (lostintangent) [inline package.json]: Can't this refer to a file in the VSIX? It would be ideal if we could ship this with the extension itself.

Comment (legomushroom) [reply package.json]: Ya, but we cannot update the schema for the older clients without deployment then.

File: package.json [hunk 2]
```diff
@@ -334,6 +334,16 @@
...
         "category": "GistPad"
+      },
+      {
+        "command": "gistpad.pasteScreenshot",
+        "title": "Paste Screenshot",
+        "category": "GistPad"
+      },
+      {
+        "command": "gistpad.addPlaygroundLibrary",
+        "title": "Add library",
```
Comment (lostintangent) [inline package.json]: `Add Library`

File: package.json (hunk 1 above)
Comment (legomushroom) [reply package.json]: Oh also the serverless scenarios. We need to host the file somewhere for that.

File: package.json (hunk 2 above)
Comment (legomushroom) [reply package.json]: Fixed.

File: src/webView/index.ts [hunk 3]
```diff
@@ -26,22 +31,76 @@ export class PlaygroundWebview {
     this.rebuildWebview();
   }
 
-  public updateHTML(html: string) {
+  public async updateLibraryDependencies(playgroundText: string) {
```
Comment (lostintangent) [inline src/webView/index.ts]: Should we call this `updateManifest`? Since we'll add more things to this file over time.

**Comment by lostintangent** [comment]
You should update this line to also check whether `react` is present in the `libraries` list. That way, you can have ` .js` file and include JSX in it, and we'll know to compile it with TypeScript: https://github.com/vsls-contrib/gistpad/blob/master/src/commands/playground.ts#L70. Otherwise, I don't want to always run a TS compile on every edit, for any arbitrary JS file.

**Comment by lostintangent** [comment]
Also, I think we should automatically include React if the playground's script file is `*.jsx` or `*.tsx`. That way, we fully optimize playgrounds for React scenarios, without the end-user having to touch the `playground.json` file at all.

**Comment by legomushroom** [comment]
@lostintangent 
> Also, I think we should automatically include React if the playground's script file is *.jsx or *.tsx. That way, we fully optimize playgrounds for React scenarios, without the end-user having to touch the playground.json file at all.

You mean on rename? There is no way to create the playground with jsx, right?

**Comment by lostintangent** [comment]
@legomushroom I added new “javascriptreact”
and “typescriptreact” options to the “Playground: Script Language”, so that you can choose to use .jsx or .tsx for new playgrounds. So you could either rename the file manually and/or use that setting.

**Comment by legomushroom** [comment]
@lostintangent I see, sounds good.

File: src/webView/index.ts (hunk 3 above)
Comment (legomushroom) [reply src/webView/index.ts]: Renamed.

**Comment by legomushroom** [comment]
Addressed the jsx compilation feedback and added auto injection of the `react` dependency depending on the `jsreact`/`tsreact` settings.

**Pull request merged**